src/arcsf/data/tofu.py

```diff
@@ -0,0 +1,193 @@
+import math
+
+import numpy as np
+from datasets import Dataset, load_dataset
+from sklearn.model_selection import train_test_split
+
+
+def _flatten_list_of_lists(list: list[list]) -> list:
+    return sum(list, [])
+
+
+# returns the question index in the dataset, given an author number and question
+# number (within the author) (inner function)
+def _get_forget_index(
+    author: int,
+    question: int,
+    q_per_author: int,
+) -> int:
+    return question + q_per_author * author
+
+
+# As above returns the question indices in the dataset, given an
+# author number and question numbers (within the author) (inner function)
+def _get_forget_indices(
+    authors: int,
+    questions: int | list[int],
+    q_per_author: int,
+) -> int | list[int]:
+    if isinstance(questions, int):
+        return _get_forget_index(authors, questions, q_per_author)
+    if isinstance(questions, list):
+        return [
+            _get_forget_index(authors, question, q_per_author) for question in questions
+        ]
+
+
+# As above returns the question indices in the dataset, given
+# author numbers and question numbers (across dataset)
+def get_forget_indices(
+    authors: int | list[int],
+    questions: int | list[int],
+    q_per_author: int,
+) -> list[int]:
+    if isinstance(authors, int):
+        return _get_forget_indices(authors, questions, q_per_author)
+    if isinstance(authors, list):
+        out = [
+            _get_forget_indices(author, questions, q_per_author) for author in authors
+        ]
+        if isinstance(out[0], list):
+            return _flatten_list_of_lists(out)
+        return out
```

Comment:
as per `_get_forget_indices`:

```suggestion
    if isinstance(authors, int):
        return _get_forget_indices(authors, questions, q_per_author)

    out = [
        _get_forget_indices(author, questions, q_per_author) for author in authors
    ]
    if isinstance(out[0], list):
        return _flatten_list_of_lists(out)
    return out
```